apps/launch/components/MyStakingContracts/Details/ContractConfiguration.tsx

```diff
@@ -0,0 +1,258 @@
+
+const { Text } = Typography;
+
+const ShowContent = ({ isLoading, data }: { isLoading: boolean; data?: string | ReactNode }) => {
+  const { networkId } = useAppSelector((state) => state.network);
+
+  if (!networkId || isLoading) return <Skeleton.Input active size="small" />;
+  return <Text>{data || NA}</Text>;
+};
+
+const ShowNetworkAddress = ({ address }: { address: Address }) => {
+  const { networkId } = useAppSelector((state) => state.network);
+
+  if (!networkId) return null;
+  if (!address) return null;
+
+  const truncatedAddress = truncateAddress(address);
+  return (
+    <a href={`${EXPLORER_URLS[networkId]}/address/${address}`} target="_blank" rel="noreferrer">
+      {`${truncatedAddress} ${UNICODE_SYMBOLS.EXTERNAL_LINK}`}
+    </a>
+  );
+};
+
+const MaximumStakedAgents: FC<{ address: Address }> = ({ address }) => {
+  const { data, isLoading } = useMaxNumServices({ address });
+  return <ShowContent isLoading={isLoading} data={data} />;
+};
+
+const Rewards: FC<{ address: Address }> = ({ address }) => {
+  const { data, isLoading } = useRewardsPerSecond({ address });
+  return <ShowContent isLoading={isLoading} data={`${data} OLAS` || NA} />;
+};
+
+const MinimumStakingDeposit: FC<{ address: Address }> = ({ address }) => {
+  const { data, isLoading } = useGetMinimumStakingDeposit({ address });
+  return <ShowContent isLoading={isLoading} data={data} />;
+};
+
+const MinimumStakingPeriods: FC<{ address: Address }> = ({ address }) => {
+  const { isLoading: isLivenessPeriodLoading, data: livenessPeriod } = useGetLivenessPeriod({
+    address,
+  });
+  const { isLoading: isMinimumStakingPeriodsLoading, data: minimumStakingPeriods } =
+    useGetMinimumStakingPeriods({ address });
+
+  const isLoading = useMemo(
+    () => isLivenessPeriodLoading || isMinimumStakingPeriodsLoading,
+    [isLivenessPeriodLoading, isMinimumStakingPeriodsLoading],
+  );
+  const data = useMemo(() => {
+    if (!livenessPeriod) return NA;
+    if (!minimumStakingPeriods) return NA;
+
+    const minStakingDuration = Number(minimumStakingPeriods) / Number(livenessPeriod);
+    return minStakingDuration;
+  }, [livenessPeriod, minimumStakingPeriods]);
+
+  return <ShowContent isLoading={isLoading} data={data} />;
+};
+
+const MaximumInactivityPeriods: FC<{ address: Address }> = ({ address }) => {
+  const { data, isLoading } = useGetMaximumInactivityPeriods({ address });
+  return <ShowContent isLoading={isLoading} data={data} />;
+};
+
+const LivenessPeriod: FC<{ address: Address }> = ({ address }) => {
+  const { data, isLoading } = useGetLivenessPeriod({ address });
+  return <ShowContent isLoading={isLoading} data={data ? `${data} seconds` : NA} />;
+};
+
+const TimeForEmissions: FC<{ address: Address }> = ({ address }) => {
+  const { data, isLoading } = useTimeForEmissions({ address });
+  return <ShowContent isLoading={isLoading} data={data} />;
+};
+
+const AgentInstances: FC<{ address: Address }> = ({ address }) => {
```

Comment:
can you please rename this to `NumAgentInstances` as well as the `useAgentInstances` hook? currently it sounds like it returns a list of instances